Kernel/Net/TCPSocket.cpp

```diff
@@ -183,25 +183,33 @@ KResult TCPSocket::send_ack(bool allow_duplicate)
 KResult TCPSocket::send_tcp_packet(u16 flags, const UserOrKernelBuffer* payload, size_t payload_size)
 {
     const bool has_options = flags & TCPFlags::SYN;
-    const size_t options_size = has_options ? sizeof(TCPOptionMSS) : 0;
+    const size_t options_size = has_options ? (sizeof(TCPOptionMSS) + sizeof(TCPOptionWindowScale) + sizeof(TCPOptionEnd)) : 0;
     const size_t header_size = sizeof(TCPPacket) + options_size;
     const size_t buffer_size = header_size + payload_size;
     auto buffer = NetworkByteBuffer::create_zeroed(buffer_size);
     auto& tcp_packet = *(TCPPacket*)(buffer.data());
     VERIFY(local_port());
     tcp_packet.set_source_port(local_port());
     tcp_packet.set_destination_port(peer_port());
-    tcp_packet.set_window_size(NumericLimits<u16>::max());
+    tcp_packet.set_window_size(m_window_size);
     tcp_packet.set_sequence_number(m_sequence_number);
     tcp_packet.set_data_offset(header_size / sizeof(u32));
     tcp_packet.set_flags(flags);
 
+    auto now = kgettimeofday();
+
     if (flags & TCPFlags::ACK) {
         m_last_ack_number_sent = m_ack_number;
-        m_last_ack_sent_time = kgettimeofday();
+        m_last_ack_sent_time = now;
         tcp_packet.set_ack_number(m_ack_number);
     }
 
+    if (m_window_size <= NumericLimits<u16>::max() - window_increase_step
+        && m_last_window_increase < kgettimeofday() - window_increase_interval) {
```

Comment:
this conditional is looking really chonky, would it be possible to put it on a diet?

Reply:
possibly replace `kgettimeofday` with `now`?